classes/CryptedConfiguration.php

```diff
@@ -0,0 +1,200 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/**
+ * Class CryptedConfiguration.
+ */
+class CryptedConfiguration extends Configuration
+{
+    /**
+     * @var PhpEncryption Instance of PhpEncryption
+     */
+    private static $phpEncryption = null;
+
+    /**
+     * Get instance of PhpEncryption
+     *
+     * @return PhpEncryption
+     */
+    public static function getPhpEncryption()
+    {
+        if (self::$phpEncryption === null) {
+            self::$phpEncryption = new PhpEncryption(_NEW_COOKIE_KEY_);
+        }
+
+        return self::$phpEncryption;
+    }
+
+    /**
+     * Get a single configuration value (in one language only).
+     *
+     * @param string $key Key wanted
+     * @param int $idLang Language ID
+     *
+     * @return string|bool Value
+     */
+    public static function get($key, $idLang = null, $idShopGroup = null, $idShop = null, $default = false)
+    {
+        $rawValue = parent::get($key, $idLang, $idShopGroup, $idShop, $default);
+        if (empty($rawValue) === true) {
+            return $rawValue;
+        }
+        $value = self::getPhpEncryption()->decrypt($rawValue);
+
+        return $value;
+    }
+
+    /**
+     * Get global value.
+     *
+     * @param string $key Configuration key
+     * @param int|null $idLang Language ID
+     *
+     * @return string
+     */
+    public static function getGlobalValue($key, $idLang = null)
+    {
+        return CryptedConfiguration::get($key, $idLang, 0, 0);
+    }
+
+    /**
+     * Get a single configuration value for all shops.
+     *
+     * @param string $key Key wanted
+     * @param int $idLang
+     *
+     * @return array Values for all shops
+     */
+    public static function getMultiShopValues($key, $idLang = null)
+    {
+        $shops = Shop::getShops(false, null, true);
+        $resultsArray = [];
+        foreach ($shops as $idShop) {
+            $resultsArray[$idShop] = CryptedConfiguration::get($key, $idLang, null, $idShop);
+        }
```

Comment:
I think we should rely on the parent class to get the data and just add the decryption layer, WDYT?
```suggestion
        $resultsArray = parent::getMultiShopValues($key, $idLang);
        foreach ($resultsArray as $idShop => $result) {
            if (empty($result) {
                continue;
            }
            $resultsArray[$idShop] = self::getPhpEncryption()->decrypt($rawValue);
        }
```